substrate/frame/identity/src/lib.rs

```diff
@@ -922,6 +961,129 @@ pub mod pallet {
+			let id = <IdentityOf<T>>::take(&who).ok_or(Error::<T>::NotNamed)?;
+			let registrars = id.judgements.len() as u32;
+			#[allow(deprecated)]
+			let fields = id.info.additional() as u32;
+			let (subs_deposit, sub_ids) = <SubsOf<T>>::take(&who);
+			// check witness data
+			let actual_subs = sub_ids.len() as u32;
+			for sub in sub_ids.iter() {
+				<SuperOf<T>>::remove(sub);
+			}
+
+			// unreserve any deposits
+			let deposit = id.total_deposit().saturating_add(subs_deposit);
+			let err_amount = T::Currency::unreserve(&who, deposit);
+			debug_assert!(err_amount.is_zero());
+
+			// Finally, call the handler.
+			T::ReapIdentityHandler::on_reap_identity(&who, fields, actual_subs)?;
+			Self::deposit_event(Event::IdentityReaped { who });
+			Ok(Some(T::WeightInfo::reap_identity(registrars, actual_subs)).into())
+		}
+
+		/// Update the deposits held by `target` for its identity info.
+		///
+		/// Parameters:
+		/// - `target`: The account for which to update deposits.
+		///
+		/// May be called by any signed origin.
+		#[pallet::call_index(16)]
+		#[pallet::weight(T::WeightInfo::poke_deposit())]
+		pub fn poke_deposit(origin: OriginFor<T>, target: AccountIdLookupOf<T>) -> DispatchResult {
+			// No locked check: used for migration.
+			// anyone or root (so that the system can call it for identity migration)
+			let _ = ensure_signed_or_root(origin)?;
+			let target = T::Lookup::lookup(target)?;
+
+			// Identity Deposit
+			ensure!(IdentityOf::<T>::contains_key(&target), Error::<T>::NoIdentity);
+			let new_id_deposit = IdentityOf::<T>::try_mutate(
+				&target,
+				|registration| -> Result<BalanceOf<T>, DispatchError> {
+					let reg = registration.as_mut().ok_or(Error::<T>::NoIdentity)?;
+					// Calculate what deposit should be
+					#[allow(deprecated)]
+					let field_deposit = T::FieldDeposit::get()
+						.saturating_mul(BalanceOf::<T>::from(reg.info.additional() as u32));
+					let new_id_deposit = T::BasicDeposit::get().saturating_add(field_deposit);
+
+					// Update account
+					Self::rejig_deposit(&target, reg.deposit, new_id_deposit)?;
+
+					reg.deposit = new_id_deposit;
+					Ok(new_id_deposit)
+				},
+			)?;
+
+			// Subs Deposit
+			let new_subs_deposit = SubsOf::<T>::try_mutate(
+				&target,
+				|(current_subs_deposit, subs_of)| -> Result<BalanceOf<T>, DispatchError> {
+					let new_subs_deposit = Self::subs_deposit(subs_of.len() as u32);
+					Self::rejig_deposit(&target, *current_subs_deposit, new_subs_deposit)?;
+					*current_subs_deposit = new_subs_deposit;
+					Ok(new_subs_deposit)
+				},
+			)?;
+
+			Self::deposit_event(Event::DepositUpdated {
+				who: target,
+				identity: new_id_deposit,
+				subs: new_subs_deposit,
+			});
+			Ok(())
+		}
+
+		/// Lock the pallet.
+		///
+		/// Origin must be the `LockerOrigin`.
+		#[pallet::call_index(17)]
+		#[pallet::weight(T::WeightInfo::lock_pallet())]
+		pub fn lock_pallet(origin: OriginFor<T>) -> DispatchResult {
```

Comment:
To lock the pallet, we should just remove the possibility to call this pallet from the runtime. Not requiring any special logic in this pallet here. This can be done by just removing `Call` in the `Identity` line in `construct_runtime`.

Reply:
That was my initial approach but I think that is wrong for two reasons:

1. We still want to be able to call `reap_identity`, and
2. It will result in a lot of downtime because the unlocking action will need to be done via runtime upgrade. I was going to propose that the Fellowship be the `LockerOrigin` such that it could unlock the parachain once all deposits have been migrated. We can then change it to Root.

Reply:
> 2\. such that it could unlock the parachain once all deposits have been migrated

But this will require on the users actually doing this? Sounds like it takes quite some time?

Reply:
No, once the parachain launches where we want the deposits to migrate to, the `ReapOrigin` would be set to `EnsureSigned` so that anyone could call this and migrate every account. See the docs on the [Rococo Config](https://github.com/paritytech/polkadot-sdk/pull/1814/files#diff-da220c6c064cc047fef5d3b4a594ae30b8bbf48a29b740742bc9858285c7ba57R623).

Since the `IdentityInfo` and other state will be in the parachain's genesis (with the pallet locked at genesis), we can very quickly migrate all the deposits to the parachain and unlock the pallet there. Then we just remove the pallet from the Relay Chain entirely in the next runtime upgrade.